UnifiedApi/Client/Services/Submitter/Service.cs

```diff
@@ -281,50 +281,65 @@ public string SessionId
 
     while (missing.Count != 0)
     {
-      foreach (var bucket in missing.Batch(10000))
+      foreach (var bucket in missing.Batch(500))
       {
-        try
-        {
-          var partialResults = SessionService.TryGetResults(bucket);
+        var resultStatusCollection = SessionService.GetResultStatus(bucket);
 
-          foreach (var (taskId, bytesArray) in partialResults)
+        foreach (var taskId in resultStatusCollection.IdsReady.Select(tuple => tuple.Item1))
+        {
+          try
           {
             responseHandler(taskId,
-                            bytesArray);
+                            SessionService.TryGetResult(taskId,
+                                                        false,
+                                                        CancellationTokenSource.Token));
+          }
+          catch (Exception e)
+          {
+            errorHandler(taskId,
+                         e.Message + e.StackTrace);
           }
+        }
 
-          results.AddRange(partialResults);
+        resultsCount += resultStatusCollection.IdsReady.Count();
```

Comment:
You can directly subtract `missing.Count()` from its original value to get the number of results. No need to call `Count()` on the result collections.